src/common/components/CustomIcon/SchoolIcon/BaseIcon.tsx

```diff
@@ -0,0 +1,24 @@
+import { CustomIcon, type CustomIconProps } from "../CustomIcon";
+import { NTUIcon } from "@/common/components/CustomIcon/SchoolIcon/NTUIcon";
+import { NUSIcon } from "@/common/components/CustomIcon/SchoolIcon/NUSIcon";
+import { SMUIcon } from "@/common/components/CustomIcon/SchoolIcon/SMUIcon";
+
+export interface SchoolIconProps extends CustomIconProps {
+  school: "NTU" | "SMU" | "NUS";
+}
+
+export const SchoolIcon = (props: SchoolIconProps) => {
+  const { school, ...CustomIconProps } = props;
+  const svgs = {
+    SMU: SMUIcon,
+    NUS: NUSIcon,
+    NTU: NTUIcon,
+  };

```

Comment:
I think we can move this out of this function since it's not dynamic. You can just put it in the same file, maybe in line 9!

Reply:
Roger!